network/network.go

```diff
@@ -711,6 +762,21 @@ func (n *network) Dispatch() error {
 			n.log.Debug("error during server accept: %s", err)
 			return err
 		}
+
+		// We pessimistically drop an incoming connection if the remote
+		// address is found in connectedIPs, myIPs, or peerAliasIPs.
+		// This protects our node from spending CPU cycles on TLS
+		// handshakes to upgrade connections from existing peers.
+		// Specifically, this can occur when one of our existing
+		// peers attempts to connect to one our IP aliases (that they
+		// aren't yet aware is an alias).
+		addr := conn.RemoteAddr().String()
+		if !n.upgradeIncoming(addr) {
```

Comment:
I added a pessimistic connection drop if the `remoteAddr` is in `connectedIPs`, `myIPs`, or `peerAliasIPs`. This is a defensive mechanism on the `dialee` side to drop incoming duplicate connections from existing peers (opened either accidentally or purposely) before the TLS handshake is performed (on ours or the client side). Inadvertently, it also switches the "peer alias" usage to mostly be on the server side instead of the client side (accepting that we will probably close the connection before the `dialer` can upgrade). Happy to discuss whether you agree this is the right direction @danlaine @aaronbuchwald @StephenButtolph.

If our IP has changed but we still maintain connections with peers from our old IP, it is possible that existing peers will attempt to dial our new IP as if we are a new peer. If we are already connected, we will now close the connection here and they will fail to upgrade the connection (instead retrying after some backoff). This could prevent the `dialer` from marking our new IP as an alias but this would already occur if we drop the connection before they can complete the handshake (when we realize we are already connected to them).

If an existing peer is connecting from an alias IP, the existing "alias changes" protect us from continuing to upgrade their incoming connection requests to check if they are an existing peer.

Reply:
@aaronbuchwald the universal backoff tracking in #749 will be a GREAT complement to this approach.